### PR TITLE
feat: allow reading and watching global attributes

### PR DESCRIPTION
**Pull request by jtmalinowski**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -308,7 +345,19 @@ const SplunkRum: SplunkOtelWebType = {
     }
 
     _errorInstrumentation.report('SplunkRum.error', args);
-  }
+  },
+
+  _experimental_addEventListener(name: SplunkOtelWebEventType, callback: (event: never) => void): void {
```
Comment (t2t2) [inline src/index.ts:350]: What's the point of having internal event target methods also have experimental prefix when it's publicly only exposed here? Could get rid of the custom `NativeEventTarget`

File: src/EventTarget.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  _experimental_addEventListener(name: SplunkOtelWebEventType.GlobalAttributesChanged, callback: (event: GlobalAttributesChangedEvent) => void);
+
+  _experimental_removeEventListener(name: SplunkOtelWebEventType.SessionChanged, callback: (event: SessionIdChangedEvent) => void);
+  _experimental_removeEventListener(name: SplunkOtelWebEventType.GlobalAttributesChanged, callback: (event: GlobalAttributesChangedEvent) => void);
+}
+
+export class NativeEventTarget implements SplunkOtelWebEventTarget {
+  // we're using DOM API to save on bytes a little bit
+  // another benefit is
+  private readonly target = new EventTarget;
```
Comment (t2t2) [inline src/EventTarget.ts:55]: Bad compatibility in IE/safari <14 https://developer.mozilla.org/en-US/docs/Web/API/EventTarget#browser_compatibility

File: src/EventTarget.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  _experimental_addEventListener(name: SplunkOtelWebEventType.SessionChanged, callback: (event: SessionIdChangedEvent) => void);
+  _experimental_addEventListener(name: SplunkOtelWebEventType.GlobalAttributesChanged, callback: (event: GlobalAttributesChangedEvent) => void);
+
+  _experimental_removeEventListener(name: SplunkOtelWebEventType.SessionChanged, callback: (event: SessionIdChangedEvent) => void);
+  _experimental_removeEventListener(name: SplunkOtelWebEventType.GlobalAttributesChanged, callback: (event: GlobalAttributesChangedEvent) => void);
+}
+
+export class NativeEventTarget implements SplunkOtelWebEventTarget {
+  // we're using DOM API to save on bytes a little bit
+  // another benefit is
```
[the remark this answers is not in the document]
Comment (jtmalinowski) [reply src/EventTarget.ts]: fixed

File: src/index.ts (hunk 1 above)
Comment (jtmalinowski) [reply src/index.ts:350]: It's just convenient to reuse `SplunkOtelWebEventTarget` interface internally and externally.

File: src/EventTarget.ts (hunk 2 above)
Comment (jtmalinowski) [reply src/EventTarget.ts]: replaced with `HTMLDivElement`, good catch!

**Review by mhennoch — APPROVED** [review]

File: src/EventTarget.ts [hunk 4]
```diff
@@ -41,28 +41,21 @@ export class GlobalAttributesChangedEvent extends Event {
...
+export interface SplunkOtelWebEventTarget {
+  _experimental_addEventListener: InternalEventTarget['addEventListener'];
+  _experimental_removeEventListener: InternalEventTarget['removeEventListener']
+}
 
-  dispatchEvent(event: SessionIdChangedEvent | GlobalAttributesChangedEvent): void {
-    this.target.dispatchEvent(event);
-  }
+export function buildInternalEventTarget(): InternalEventTarget {
+  return document.createElement('div');
```
Comment (t2t2) [inline src/EventTarget.ts:60]: Maybe a comment for in 3 months when everyone's forgotten about this

```suggestion
  // EventTarget can't be initiated in older versions of safari and IE so use something that already implements it
  return document.createElement('div');
```

**Review by t2t2 — APPROVED** [review]

**Pull request merged**